crowbar_framework/app/models/service_object.rb

```diff
@@ -1186,6 +1190,9 @@ def apply_role(role, inst, in_queue)
           tmprole = RoleObject.find_role_by_name "#{role_name}_remove"
           unless tmprole.nil?
             elem_remove = tmprole.name
+            # save remove intention in #{@bc_name}-databag
+            databag["crowbar"]["roles_to_remove"] << "#{role_name}_remove"
```

Comment:
Hrm: shouldn't we save for which nodes the role will be removed? Otherwise, I'm not sure it's very useful. My main worry was that we lose the reference of the nodes that are removed when apply_role fails somehow. So not the reference of the roles, but really the nodes.